packages/ember-application/lib/system/application.js

```diff
@@ -261,42 +263,19 @@ var Application = Namespace.extend(DeferredMixin, {
       this.$ = jQuery;
     }
 
+    // Create subclass of Ember.Router for this Application instance.
+    // This is to ensure that someone reopening `App.Router` does not
+    // tamper with the default `Ember.Router`.
+    this.Router = Router.extend();
```

Comment:
In the 2.0 RFC it now states:
> While globals-based apps will continue to work in 2.0, we may introduce new features that rely on either Ember CLI or ES6 modules. You should begin moving your app to Ember CLI as soon as possible.

So globals mode is probably something that should be flagged separately. Maybe require it to be explicitly defined using a flag during App creation? (and default to `true` in 1.x and `false` in 2.x)